src/utils/memoize.ts

```diff
@@ -0,0 +1,19 @@
+import { memoize } from 'lodash-es'
+export function memoizePromise<T extends (...args: Args) => PromiseLike<any>, Args extends any[]>(
+    f: T,
+    resolver: (...args: Args) => any = ((a: any) => a) as any,
+) {
+    const memorizedFunction = memoize(
+        (async function(...args: Args) {
+            try {
+                // ? DO NOT remove "await" here
+                return await f(...args)
+            } catch (e) {
+                memorizedFunction.cache.delete(resolver(...args))
```

Comment:
resolver is possibly undefined here?... Typescript is not reporting this so I'm wondering if I got it wrong...

Reply:
It has a default value `((a: any) => a) as any`